### DISCUSSION TITLE
Nested groups

### DISCUSSION THREAD

**Discussion by EinLinuus** [tailwindlabs/tailwindcss#6034] [Help] [answered]
Hey there,

recently, I had a problem with the group utility:

```html
<button class="border rounded-lg p-2 group hover:bg-black">
   <span class="font-bold group-hover:text-white">Hello world</span>
   <span class="relative group-hover:text-white group">
      <span>[Info]</span>
      <!--  Tooltip-->
      <span class="hidden group-hover:block p-1 border bg-white text-xs rounded absolute top-full left-0 text-black whitespace-nowrap">I am a text</span>
   </span>
</button>
```
_Not my real code, but you see the problem._

My goal is to have the color-change effect when hovering over the button, but the tooltip should only be visible when hovering over the [Info] span.

But as you can see (https://play.tailwindcss.com/Ti828DAb1Q), the tooltip is visible when I hover anywhere over the button. That makes sense and I know why, but I need a solution here.

I tried different TailwindCSS third-party plugins, but they were incompatible with the JIT compiler or didn't work at all.

**Now, this is my first contribution to an open-source project, so I'm not sure how everything works and what exactly I have to do.**
I tried to implement this on my own and added a new utility, `group-scoped`:

```html
<button class="border rounded-lg p-2 group hover:bg-black">
   <span class="font-bold group-hover:text-white">Hello world</span>
   <span class="relative group-hover:text-white group-scoped">
      <span>[Info]</span>
      <span class="hidden group-scoped-hover:block p-1 border bg-white text-xs rounded absolute top-full left-0 text-black whitespace-nowrap">I am a text</span>
   </span>
</button>
```

Here is an example of the CSS output for `group-scoped-hover:block`:
```css
.group-scoped:hover > .group-scoped-hover\:block {
  display: block;
}
.group-scoped:hover *:not(.group-scoped) .group-scoped-hover\:block {
  display: block;
}
```

**My question:** What can I do next? How can I use this in my projects - do I have to create a TailwindCSS plugin? Can I just create a PR to add this to the core - am I even allowed to do that?

In my opinion, it would be awesome to have something like this in the core, I also found others having this problem.
Where can I suggest that? Do I just have to create a pull request to the tailwindcss master branch?

Thanks in advance 🙏

**Comment by EinLinuus** [comment]
If anyone has the same problem and needs a solution, I created a plugin:
https://www.npmjs.com/package/tailwindcss-scoped-groups

**Comment by reinink** [comment, marked as answer]
Hey! Good news — we just added nested group and multiple peer support using variant modifiers in Tailwind CSS v3.2 🥳

You can read about it in our [announcement blog post](https://tailwindcss.com/blog/tailwindcss-v3-2#nested-group-and-multiple-peer-support-using-variant-modifiers), but for quick reference here’s what it looks like:

```html
<div class="group/sidebar ...">
  <!-- ... -->
  <div class="group/navitem ...">
    <a
      href="#"
      class="opacity-50 group-hover/sidebar:opacity-75 group-hover/navitem:bg-black/75"
    >
      <!-- ... -->
    </a>
  </div>
  <!-- ... -->
</div>
```

This lets you give each group a clear name that makes sense for that context on the fly, and Tailwind will generate the necessary CSS to make it work.

**Reply by sebastienbarre** [reply]
Great, thanks, I was finally able to get rid of (the quite useful back then) `tailwindcss-labeled-groups` NPM

**Reply by EinLinuus** [reply]
That's great news, thanks!

**Reply by mati-1** [reply]
thanks bro

**Reply by sridhar-neartekpod** [reply]
Nice

**Reply by NotNikita** [reply]
Great, thanks

**Reply by Jervx** [reply]
Thanks ✨

**Reply by hexxt-git** [reply]
epic

**Reply by its-tim-lee** [reply]
but does this support custom prefix?

**Reply by wongjn** [reply]
[Yep!](https://play.tailwindcss.com/YHnDiQgssm)

**Reply by its-tim-lee** [reply]
> [Yep!](https://play.tailwindcss.com/YHnDiQgssm)


@wongjn 
I knew it works in v4, but it seems doesn't work in v3

**Reply by wongjn** [reply]
[Seems to work in v3 too](https://play.tailwindcss.com/Lmf51ji2Rw).

**Reply by paulinevos** [reply]
Late to the party but this is an awesome feature

**Comment by moladukes** [comment]
Tailwind keeps winning
